x-pack/plugins/security/public/account_management/api_keys_page/create_api_key_flyout.tsx

```diff
@@ -0,0 +1,273 @@
+      onSubmitSuccess: onSuccess,
+      onSubmitError: onError,
+      validate,
+      defaultValues,
+    },
+    [services.http]
+  );
+  const nameInput = useRef<HTMLInputElement>(null);
+
+  return (
+    <FormFlyout
+      title={i18n.translate('xpack.security.accountManagement.createApiKey.title', {
+        defaultMessage: 'Create API key',
+      })}
+      submitButtonText={i18n.translate(
+        'xpack.security.accountManagement.createApiKey.submitButton',
+        {
+          defaultMessage: '{isSubmitting, select, true{Creating API key…} other{Create API key}}',
+          values: { isSubmitting: form.isSubmitting },
+        }
+      )}
+      isLoading={form.isSubmitting}
+      initialFocus={nameInput}
+      onSubmit={eventHandlers.onSubmit}
+      onClose={onClose}
+    >
+      {form.submitError && (
+        <>
+          <EuiCallOut
+            title={i18n.translate('xpack.security.accountManagement.createApiKey.errorMessage', {
+              defaultMessage: 'Could not create API key',
+            })}
+            color="danger"
+          >
+            {(form.submitError as any).body?.message || form.submitError.message}
+          </EuiCallOut>
+          <EuiSpacer />
+        </>
+      )}
+      <EuiForm
+        component="form"
+        isInvalid={form.isInvalid}
+        error={Object.values(form.errors)}
+        invalidCallout={!form.submitError && form.isSubmitted ? 'above' : 'none'}
+        {...eventHandlers}
+      >
+        <EuiFormRow
+          label={i18n.translate('xpack.security.accountManagement.createApiKey.nameLabel', {
+            defaultMessage: 'Name',
+          })}
+          helpText="What is the API key used for?"
+          error={form.errors.name}
+          isInvalid={!!form.errors.name}
+        >
+          <EuiFieldText
+            name="name"
+            defaultValue={form.values.name}
+            isInvalid={!!form.errors.name}
+            inputRef={nameInput}
+          />
+        </EuiFormRow>
+
+        <EuiSpacer />
+        <EuiSwitch
+          id="apiKeyCustom"
+          label={i18n.translate(
+            'xpack.security.accountManagement.createApiKey.customPriviligesLabel',
+            {
+              defaultMessage: 'Grant subset of your permissions',
+            }
+          )}
+          checked={!!form.values.customPriviliges}
+          onChange={(e) => form.setValue('customPriviliges', e.target.checked)}
+        />
+        {form.values.customPriviliges && (
+          <>
+            <EuiSpacer />
+            <EuiFormRow
+              label={i18n.translate(
+                'xpack.security.accountManagement.createApiKey.roleDescriptorsLabel',
+                {
+                  defaultMessage: 'Role Descriptors',
+                }
+              )}
+              helpText="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html#security-api-create-api-key-request-body"
```

Comment:
It'd be nice if links to documentation was this easy, but we actually need to link to the proper version of the docs. In other places, we define a  [`DocumentationLinksService`](https://github.com/elastic/kibana/blob/9d3d3cdc67e692c90d3bc959984114f007156615/x-pack/plugins/security/public/management/role_mappings/documentation_links.ts) to encapsulate this uglyness.

I'd like to see something like that included here as well. If you wouldn't mind, we could pull up that role mappings doc links service to a "sharable" location within the security plugin, and add this doc link to that service so that we have a single reference point in the future.

Reply:
Haha, good catch. This was not supposed to stay like that. 🙈
I'll turn this into a proper link and copy.

Reply:
I've added a `useDocLink` hook and `DocLink` component that handles this logic.